### PR TITLE
Refactor cosypose wrapper

### PR DESCRIPTION
**Pull request by MedericFourmy**
In CosyPoseWrapper class
- add a `model_type` "pbr" and "synth+real" model weights
- list available models ids for both "pbr" and "synth+real" model weights 
- does not change current default behavior (load "pbr" models)
- remove unused argument gpu_renderer
- simplify unused function `inference`

### CODE REVIEW

File: happypose/pose_estimators/cosypose/cosypose/utils/cosypose_wrapper.py [hunk 1]
```diff
@@ -32,62 +33,93 @@
...
+        """
+        Return CosyPose detector and pose estimator objects for a given dataset.
+
+        :param dataset_name: str, name of the dataset on which model was trained, hope|tless|ycbv
+        :param model_type: str, type of NN model (depends on training data), hope|tless|ycbv
+        :param renderer_type: str, which renderer to use, "panda3d" and "bullet" supported
+        :param n_workers: int, number of workers used in the renderer
+        :param model_type: str, what training data was used, "pbr" and "synth+real" supported
+        :return: tuple (Detector,PoseEstimator)
+        """
```
Comment (Kotochleb) [inline happypose/pose_estimators/cosypose/cosypose/utils/cosypose_wrapper.py]: Which docstring type is the dominating one in happypose?

Comment (MedericFourmy) [reply happypose/pose_estimators/cosypose/cosypose/utils/cosypose_wrapper.py]: It seems like the cosypose inherited part of the code lacks docstring while the megapose one seems to follow something in between the google and numpydoc style (if you know what this is):

https://github.com/agimus-project/happypose/blob/388a0bff0e4f4c3ab51b599ac804d4c2583ff007/happypose/pose_estimators/megapose/inference/detector.py#L78 

I'll edit the new comments to fit this style for now.

Comment (Kotochleb) [reply happypose/pose_estimators/cosypose/cosypose/utils/cosypose_wrapper.py]: It looks like google with lines separating. I think we could stick to google. But maybe @nim65s could you relate to this topic?

Comment (nim65s) [reply happypose/pose_estimators/cosypose/cosypose/utils/cosypose_wrapper.py]: this is sphinx: https://www.sphinx-doc.org/en/master/usage/domains/python.html#info-field-lists

Comment (Kotochleb) [reply happypose/pose_estimators/cosypose/cosypose/utils/cosypose_wrapper.py]: @nim65s I meant the docstrigns in megapose @MedericFourmy sent link to. I know the one he used in this PR is sphinx, but the problem is that the rest of the codebase seems to be using google style

File: happypose/pose_estimators/cosypose/cosypose/utils/cosypose_wrapper.py [hunk 2]
```diff
@@ -32,62 +36,98 @@
...
-        inputs:
-        - dataset_name: hope|tless|ycbv
-        - object_dataset: None or already existing rigid object dataset. If None, will use dataset_name
-        to build one
-        - n_workers: how many processes will be spun in the batch renderer
-        - renderer_type: 'panda3d'|'bullet'
+        Args:
+        ----
+            dataset_name: str, name of the dataset on which model was trained, hope|tless|ycbv
+            model_type: str, type of NN model (depends on training data), hope|tless|ycbv
```
Comment (Kotochleb) [inline happypose/pose_estimators/cosypose/cosypose/utils/cosypose_wrapper.py]:
```suggestion
            model_type: str, type of NN model (depends on training data), pbr|synth+real
```

File: happypose/pose_estimators/cosypose/cosypose/utils/cosypose_wrapper.py [hunk 3]
```diff
@@ -32,62 +36,98 @@
...
-        detector = load_detector(detector_run_id, device)
+    def get_model(
+        self, dataset_name, model_type, n_workers, renderer_type
+    ) -> Tuple[Detector, PoseEstimator]:
+        """Return CosyPose detector and pose estimator objects for a given dataset.
+
+        Args:
+        ----
+            dataset_name: str, name of the dataset on which model was trained, hope|tless|ycbv
+            model_type: str, type of NN model (depends on training data), hope|tless|ycbv
```
Comment (Kotochleb) [inline happypose/pose_estimators/cosypose/cosypose/utils/cosypose_wrapper.py]:
```suggestion
            model_type: str, type of NN model (depends on training data), synth+real 
```

File: happypose/pose_estimators/cosypose/cosypose/utils/cosypose_wrapper.py [hunk 4]
```diff
@@ -139,26 +179,20 @@ def load_pose_models(
...
-            final_preds, all_preds = self.pose_predictor.run_inference_pipeline(
-                observation,
-                detections=detections,
-                run_detector=run_detector,
-                data_TCO_init=None,
-                n_coarse_iterations=0,
-                n_refiner_iterations=4,
-            )
+    def inference(self, observation: ObservationTensor):
+        """Example of how to un inference with the loaded models.
```
Comment (Kotochleb) [inline happypose/pose_estimators/cosypose/cosypose/utils/cosypose_wrapper.py]:
```suggestion
        """Example of how to use inference with the loaded models.
```

**Review by Kotochleb — CHANGES_REQUESTED** [review]

**Review by Kotochleb — APPROVED** [review]

**Pull request merged**